### ISSUE TITLE
apparently a bug in asyncssh/asyncssh/known_hosts.py

### ISSUE THREAD

**Issue by evilkost** [ronf/asyncssh#14]
Hi, I'm quite interested in your project, but unfortunately I've failed to run even simplest example with the following error: 
```
Exception in callback SSHConnection.connection_made(<_SelectorSoc...e, bufsize=0>>)
handle: <Handle SSHConnection.connection_made(<_SelectorSoc...e, bufsize=0>>)>
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/events.py", line 120, in _run
    self._callback(*self._args)
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 334, in connection_made
    self._connection_made()
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 1505, in _connection_made
    self._peer_addr, self._port)
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/known_hosts.py", line 142, in match_known_hosts
    host_keys, ca_keys, revoked_keys = _match_entries(entries, host, addr, port)
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/known_hosts.py", line 113, in _match_entries
    if entry.matches(host, addr, ip):
TypeError: matches() takes 3 positional arguments but 4 were given
Exception in callback _SelectorSocketTransport._read_ready()
handle: <Handle _SelectorSocketTransport._read_ready()>
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/events.py", line 120, in _run
    self._callback(*self._args)
  File "/usr/lib/python3.4/asyncio/selector_events.py", line 669, in _read_ready
    self._protocol.data_received(data)
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 349, in data_received
    while self._inpbuf and self._recv_handler():
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 430, in _recv_version
    self._send_kexinit()
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 642, in _send_kexinit
    host_key_algs = NameList(self._get_server_host_key_algs())
  File "/home/kost/.virtualenvs/copr-py3/lib/python3.4/site-packages/asyncssh/connection.py", line 1550, in _get_server_host_key_algs
    return self._server_host_key_algs
AttributeError: 'SSHClientConnection' object has no attribute '_server_host_key_algs'
```
It looks like that problem is here: https://github.com/ronf/asyncssh/blob/master/asyncssh/known_hosts.py#L31 and  #L56:
PlainHost: def matches(self, host, addr, ip)
HashedHost: def  matches(self, host, addr)
And  line https://github.com/ronf/asyncssh/blob/master/asyncssh/known_hosts.py#L113
causes TypeError

**Comment by ronf** [comment]
Thanks very much for the bug report. I’m sorry that you ran into this. It looks like support for hashed known_host entries got broken in the 1.0 release when I added support for authorized_keys files. I’ve included a patch you can try below if you like, and I’ll work on getting this merged into a new release as soon as possible. I was planning to produce a 1.1.1 release today with one other bug fix an a small improvement in the new SFTP support, and I'll include this as well.

If you’d like to try applying the fix yourself, here it is. There are actually two issues that were introduced in the refactoring that I missed:

```diff
--- a/asyncssh/known_hosts.py
+++ b/asyncssh/known_hosts.py
@@ -51,9 +51,10 @@ class HashedHost:
                                  magic) from None
 
     def _match(self, value):
-        return hmac.new(self._salt, value, sha1).digest() == self._hosthash
+        hosthash = hmac.new(self._salt, value.encode(), sha1).digest()
+        return hosthash == self._hosthash
 
-    def matches(self, host, addr):
+    def matches(self, host, addr, ip):
         return (host and self._match(host)) or (addr and self._match(addr))
 
 
```

**Comment by ronf** [comment]
This should now be fixed in the 1.1.1 release.

**Comment by evilkost** [comment]
Great, thank you for the fast fix )
